lib/stack_master/prompter.rb

```diff
@@ -1,11 +1,17 @@
 module StackMaster
   module Prompter
     def ask?(question)
-      StackMaster.stdout.print question
-      answer = if ENV['STUB_AWS']
-        ENV['ANSWER']
+      answer = if StackMaster.interactive?
+        StackMaster.stdout.print question
+        if StackMaster.stdin.tty? && StackMaster.stdout.tty?
+          StackMaster.stdin.getch.chomp
+        else
+          StackMaster.stdout.puts
+          StackMaster.stdout.puts "STDOUT or STDIN was not a TTY. Defaulting to no. To force yes use -f"
+          'n'
+        end
       else
-        STDIN.getch.chomp
+        ENV.fetch('ANSWER') { 'y' }
```

Comment:
The default without `-f` is no, with `-f` yes. ANSWER is used by our test suite to fake an answer. It might be cleaner if we introduced `StackMaster.default_non_interactive_answer` and simply stubbed it in integration tests instead.